CMakeLists.txt

```diff
@@ -0,0 +1,34 @@
+cmake_minimum_required (VERSION 3.12)
```

Comment:
what's the difference between 3.12 and 3.10?

Reply:
Functionally nothing probably.